warden/x/act/keeper/actions.go

```diff
@@ -29,7 +29,70 @@ func (approvers ApproversEnv) Get(name string) (object.Object, bool) {
 	return object.FALSE, true
 }
 
-var _ shield.Environment = ApproversEnv{}
+// ActionApprovedVotesEnv is an environment that resolves action positive votes addresses to true.
+type ActionApprovedVotesEnv []*types.ActionVote
+
+// Get implements positive action vote evaluator.Environment.
+func (votes ActionApprovedVotesEnv) Get(name string) (object.Object, bool) {
+	for _, s := range votes {
+		if s.Participant == name && s.VoteType == types.ActionVoteType_VOTE_TYPE_APPROVED {
+			return object.TRUE, true
+		}
+	}
+	return object.FALSE, true
+}
+
+// ActionRejectedVotesEnv is an environment that resolves action negative votes addresses to true.
+type ActionRejectedVotesEnv []*types.ActionVote
+
+// Get implements negative action vote evaluator.Environment.
+func (votes ActionRejectedVotesEnv) Get(name string) (object.Object, bool) {
+	for _, s := range votes {
+		if s.Participant == name && s.VoteType == types.ActionVoteType_VOTE_TYPE_REJECTED {
+			return object.TRUE, true
+		}
+	}
+	return object.FALSE, true
+}
+
+// TryExecuteVotedAction checks if the action's expression is satisfied and stores the
+// result in the database.
+func (k Keeper) TryExecuteVotedAction(ctx context.Context, act *types.Action) error {
```

Comment:
I'd like to see this split into two:

- TryExecuteAction (called when someone adds an approval)
- TryRejectAction (called when someone adds a rejection)

wdyt?